### PR TITLE
cmd: import snapd-xdg-open and integrate with our infrastructure

### PR DESCRIPTION
**Pull request by morphis**
Taken from revision 9382ccda6a5f870a254e378f3c9b54b8af27fb88 of
repository at https://github.com/snapcore/snapd-xdg-open

See https://forum.snapcraft.io/t/integrate-snapd-xdg-open-into-snapd-repository/100/ for more details

### CODE REVIEW

File: data/dbus/com.canonical.SafeLauncher.service.in [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[D-BUS Service]
+Name=com.canonical.SafeLauncher
```
Comment (zyga) [inline data/dbus/com.canonical.SafeLauncher.service.in]: I really really wish we could change this name.

File: cmd/configure.ac [hunk 2]
```diff
@@ -109,6 +109,12 @@ AS_IF([test "x$enable_apparmor" = "xyes"], [
 PKG_CHECK_MODULES([LIBUDEV], [libudev])
 PKG_CHECK_MODULES([UDEV], [udev])
 
+PKG_CHECK_MODULES([GIO], [gio-2.0])
```
Comment (zyga) [inline cmd/configure.ac]: We should make this part optional for deeply embedded environments.

File: data/dbus/com.canonical.SafeLauncher.service.in (hunk 1 above)
Comment (morphis) [inline data/dbus/com.canonical.SafeLauncher.service.in]: I saw there are some discussions around having something io.snapcraft.* based. I am fine with that and we can introduce this as a variant in another PR and migrate to it over time.

File: cmd/configure.ac (hunk 2 above)
Comment (morphis) [inline cmd/configure.ac]: Sounds good.

Comment (morphis) [inline cmd/configure.ac]: Added

**Comment by niemeyer** [comment]
@morphis Now that we have your help, I think it's time to fix this in the proper way. I'll follow through in the forum.

**Pull request closed without merging**